.circleci/config.yml

```diff
@@ -12,7 +12,7 @@ jobs:
       - NUMPY_VERSION: 'min'
       - SCIPY_VERSION: 'min'
       - MATPLOTLIB_VERSION: 'min'
-      - CYTHON_VERSION: 'min'
+      - CYTHON_VERSION: 'latest'
```

Comment:
The fix in this PR does not work with the Cython 0.28.5. I think updating this CI instance to use the latest Cython is okay because we build all our wheels with the latest stable version of Cython.

Reply:
It looks appropriate to me.

Eventually, we could pin this to `<3.0` as  Cython 3.0 might change the behavior of the implementations.

Reply:
> The fix in this PR does not work with the Cython 0.28.5. I think updating this CI instance to use the latest Cython is okay because we build all our wheels with the latest stable version of Cython.

We should rather change the Cython version in the `_min_dependencies.py` file instead. This is the centralized source of truth for our versions. There might be documentation to update as well.

> Eventually, we could pin this to <3.0 as Cython 3.0 might change the behavior of the implementations.

Why not. Probably in a dedicated PR.

Reply:
Updated PR to change `_min_dependencies.py` instead